src/bin/nydus-image/main.rs

```diff
@@ -1182,32 +1239,136 @@ impl Command {
+            };
+            // For backward compatibility with v2.1.
+            let config = Self::get_configuration(matches)?;
+            config
+                .internal
+                .set_blob_accessible(matches.get_one::<String>("bootstrap").is_none());
+            let db_strs: Vec<&str> = db_url.split("://").collect();
+            if db_strs.len() != 2 || (!db_strs[1].starts_with('/') && !db_strs[1].starts_with(':'))
+            {
+                bail!("Invalid database URL: {}", db_url);
+            }
+            match db_strs[0] {
+                "sqlite" => {
+                    let mut deduplicate: Deduplicate<SqliteDatabase> =
+                        Deduplicate::<SqliteDatabase>::new(db_strs[1])?;
+                    deduplicate.save_metadata(bootstrap_path, config, image_name, version_name)?
+                }
+                _ => {
+                    bail!("Unsupported database type: {}, please use a valid database URI, such as 'sqlite:///path/to/chunkdict.db'.", db_strs[0])
+                }
+            };
+        }
+        info!("Chunkdict metadata is saved at: {:?}", db_url);
 
+        // Connecting database and Generating chunk dictionary by algorithm "exponential_smoothing"
         let db_strs: Vec<&str> = db_url.split("://").collect();
         if db_strs.len() != 2 || (!db_strs[1].starts_with('/') && !db_strs[1].starts_with(':')) {
             bail!("Invalid database URL: {}", db_url);
         }
+        let algorithm = String::from("exponential_smoothing");
+        let _source_bootstrap_paths: Vec<PathBuf> = matches
+            .get_many::<String>("SOURCE")
+            .map(|paths| paths.map(PathBuf::from).collect())
+            .unwrap();
+
+        let (chunkdict, noise_points): (Vec<ChunkdictChunkInfo>, Vec<String>);
 
         match db_strs[0] {
             "sqlite" => {
-                let mut deduplicate: Deduplicate<SqliteDatabase> =
-                    Deduplicate::<SqliteDatabase>::new(db_strs[1])?;
-                deduplicate.save_metadata(bootstrap_path, config)?
+                let mut algorithm: deduplicate::Algorithm<SqliteDatabase> =
+                    deduplicate::Algorithm::<SqliteDatabase>::new(algorithm, db_strs[1])?;
+                let result = algorithm.chunkdict_generate()?;
+                chunkdict = result.0;
+                noise_points = result.1;
             }
             _ => {
-                bail!("Unsupported database type: {}, please use a valid database URI, such as 'sqlite:///path/to/database.db'.", db_strs[0])
+                bail!("Unsupported database type: {}, please use a valid database URI, such as 'sqlite:///path/to/chunkdict.db'.", db_strs[0])
             }
         };
-        info!("Chunkdict metadata is saved at: {:?}", db_url);
+
+        info!(
+            "The length of chunkdict is {}",
+            Vec::<ChunkdictChunkInfo>::len(&chunkdict)
+        );
+        info!("It is not recommended to use image deduplication");
```

Comment:
Why need this.

Reply:
This part is the output image name, which is used to indicate to the user that the image will not be used for the clustering algorithm

Reply:
So why do we continue to do this instead of just reporting an error?